### ISSUE TITLE
[BUG] Media hold_volume_down/hold_volume_up/release decrease/increase volume only one step

### ISSUE THREAD

**Issue by precz** [xaviml/controllerx#398]
# Bug report

## Description

When I keep decreasing volume using my IKEA E1744 device it decrease volume only one step on my yamaha receiver  

## Additional information

* Devices involved:
  * IKEA 1744 as Controller 
  * Yamaha rx-v477 as Media Player
* Integration: z2m
* AppDaemon version: 0.7.1
* ControllerX version: 4.16.0
* HACS version: 1.18.0
* Home Assistant Core version: core-2021.11.5

## AppDaemon app configuration

```yaml
yamaha:
  log_level: DEBUG
  module: controllerx
  class: E1744MediaPlayerController
  controller: sensor.pilot_glosnosci_tv_salon_action
  integration: z2m
  media_player: media_player.yamaha_receiver
  volume_steps: 40
```

## Logs

```text
2021-12-30 18:37:47.465034 INFO yamaha: 🏃 Running `Predefined (release)` now
2021-12-30 18:37:47.451189 INFO yamaha: 🎮 Button event triggered: `brightness_stop`
2021-12-30 18:37:45.749890 WARNING yamaha: There is already an action executing for `brightness_move_up`. If you want a different behaviour change `mode` parameter, the default value is `single`.
2021-12-30 18:37:45.734109 INFO yamaha: 🎮 Button event triggered: `brightness_move_up`
2021-12-30 18:37:45.433565 INFO yamaha: 🏃 Running `Predefined (hold_volume_down)` now
2021-12-30 18:37:45.398349 INFO yamaha: 🎮 Button event triggered: `brightness_move_down`
2021-12-30 18:37:42.306553 INFO yamaha: 

🤖 Service: [1mmedia_player.volume_set[0m
  - entity_id: media_player.yamaha_receiver
  - volume_level: 0.45
2021-12-30 18:37:42.303870 INFO yamaha: Previous message repeated 1 times
2021-12-30 18:37:42.259861 INFO yamaha: 🏃 Running `Predefined (hold_volume_up)` now
2021-12-30 18:37:42.246100 INFO yamaha: 🎮 Button event triggered: `brightness_move_up`
```
As in logs, I have started increasing on `18:37:42.246100` and stopped on `18:37:47.451189` but the volume was increased only once that time. I have legacy set to false on my z2m.

**Comment by xaviml** [comment]
Hi @precz 

Unfortunately, my E1744 controller does not work (I cannot pair it with Z2M). I will need to get another one to test this issue. I will keep you updated.

Could you please check this comment and let me know if this fix your issue: https://github.com/xaviml/controllerx/issues/394#issuecomment-992425097

Regards,
Xavi M.

**Comment by xaviml** [comment]
Hi @precz,

I was able to fix my Symfonisk controller, and try the E1744 controller with a Light and a Media Player. It worked flawlessly with both devices. For the Media Player I used a Google Home Mini, and these are the logs I see: 

```
2022-01-01 18:03:36.860460 INFO office_test: 🎮 Button event triggered: `brightness_move_up`
2022-01-01 18:03:36.878936 INFO office_test: 🏃 Running `Predefined (hold_volume_up)` now
2022-01-01 18:03:36.908197 INFO office_test: 
🤖 Service: media_player.volume_set
  - entity_id: media_player.livingroom_speaker
  - volume_level: 0.40
2022-01-01 18:03:37.459675 INFO office_test: 
🤖 Service: media_player.volume_set
  - entity_id: media_player.livingroom_speaker
  - volume_level: 0.50
2022-01-01 18:03:38.012287 INFO office_test: 
🤖 Service: media_player.volume_set
  - entity_id: media_player.livingroom_speaker
  - volume_level: 0.60
2022-01-01 18:03:38.563259 INFO office_test: 
🤖 Service: media_player.volume_set
  - entity_id: media_player.livingroom_speaker
  - volume_level: 0.70
2022-01-01 18:03:39.113209 INFO office_test: 
🤖 Service: media_player.volume_set
  - entity_id: media_player.livingroom_speaker
  - volume_level: 0.80
2022-01-01 18:03:39.613275 INFO office_test: 🎮 Button event triggered: `brightness_stop`
2022-01-01 18:03:39.631579 INFO office_test: 🏃 Running `Predefined (release)` now
```

I am wondering if the problem comes from the Yamaha device you have. Does this device allow receiving changes to the volume continuously? ControllerX sends a request every 500ms (default), so it is important that the device and the network can handle that. In addition, I see you set up the `volume_steps` to 40. I would recommend you using something lower like 10 or 15, otherwise, the volume steps are going to be small.

These are the specs of my setup:
- Zigbee2MQTT version: 1.22.2
- Home Assistant: 2021.12.7
- ControllerX: 4.17.0
- AppDaemon add-on version: 0.7.2
- AppDaemon version: 4.1.0
- Symfonish firmware version: 2.1.022

Regards,
Xavi M.

**Comment by xaviml** [comment]
I will close this ticket for now. If something else needs to be discussed, please open a new ticket.
